airflow/providers/imap/hooks/imap.py

```diff
@@ -219,7 +219,7 @@ def _retrieve_mails_attachments_by_name(
         self, name: str, check_regex: bool, latest_only: bool, mail_folder: str, mail_filter: str
     ) -> list:
         if not self.mail_client:
-            raise Exception("The 'mail_client' should be initialized before!")
+            raise AirflowException("The 'mail_client' should be initialized before!")
```

Comment:
```suggestion
            raise RuntimeError("The 'mail_client' should be initialized before!")
```

I guess RuntimeError would be fine in this case, when we expects that some of the values are initialised before

Reply:
Changed in all 3 cases where we check if the mail_client is initialized.